### PR TITLE
fix(reporter): TypeScript definitions for Reporter

### PR DESCRIPTION
**Pull request by ediblecode**
## Description

update gatsby type definitions for reporter. This fixes some incorrect property types and adds missing method types.

On a separate note, ideally reporter would be a separate package, which would probably also contain the reporter type defs. See https://github.com/gatsbyjs/gatsby/issues/18672#issuecomment-543384184 for a comment to that effect. I haven't addressed that here as it was a large change beyond the scope of this issue, hence just the update to the existing type defs.

## Related Issues

fixes issue #19025

### CODE REVIEW

File: packages/gatsby/index.d.ts [hunk 1]
```diff
@@ -932,32 +933,58 @@ export interface Store {
...
-    status(status: string): void
-    end: () => void
-    span: object
-  }
+  success: LogMessageType
+  verbose: LogMessageType
+  info: LogMessageType
+  warn: LogMessageType
+  log: LogMessageType
+  completeActivity(id: string, status?: string): void
```
Comment (pieh) [inline packages/gatsby/index.d.ts]: I'm not sure we want  `completeActivity` in public type definitions - this is more for internal `gatsby` usage and I would vote for not adding typing for that

File: packages/gatsby/index.d.ts [hunk 2]
```diff
@@ -932,32 +933,58 @@ export interface Store {
...
+  activityTimer(name: string, activityArgs?: ActivityArgs): ActivityTracker
+  createProgress(
+    text: string,
+    total?: number,
+    start?: number,
+    activityArgs?: ActivityArgs
+  ): ProgressActivityTracker
+  phantomActivity(
```
Comment (pieh) [inline packages/gatsby/index.d.ts]: Similar here - `phantomActivity` probably shouldn't be in the public types

File: packages/gatsby/index.d.ts [hunk 3]
```diff
@@ -932,32 +933,58 @@ export interface Store {
   replaceReducer: Function
 }
 
-type logMessageType = (format: string, ...args: any[]) => void
-type logErrorType = (message: string, error?: Error) => void
+type LogMessageType = (format: string, ...args: any[]) => void
```
Comment (pieh) [inline packages/gatsby/index.d.ts]: Args here are actually wrong, those methods on reporter (`success`, `info`, `warn`, `log`, `verbose`) only get single string argument.

File: packages/gatsby/index.d.ts [hunk 4]
```diff
@@ -932,32 +933,58 @@ export interface Store {
...
+}
+
+export type ActivityArgs = {
+  parentSpan?: Object
+  id?: string
+}
 
 export interface Reporter {
-  stripIndent: Function
+  stripIndent: TemplateTag
```
Comment (pieh) [inline packages/gatsby/index.d.ts]: This doesn't seem to be correct?
Screenshot 2019-11-05 at 02 27 50

Should this just be:

```suggestion
  stripIndent: (input: string) => string
```
or we could inline parts of https://unpkg.com/browse/@types/common-tags@1.8.0/index.d.ts to get fuller type def?

File: packages/gatsby/index.d.ts (hunk 1 above)
Comment (ediblecode) [reply packages/gatsby/index.d.ts]: thanks, yeah I did wonder about that, will remove.

File: packages/gatsby/index.d.ts (hunk 2 above)
Comment (ediblecode) [reply packages/gatsby/index.d.ts]: Likewise, will remove

File: packages/gatsby/index.d.ts (hunk 4 above)
Comment (ediblecode) [reply packages/gatsby/index.d.ts]: Probably easier to just use `(input: string) => string` as you've suggested I think. Thanks

File: packages/gatsby/index.d.ts (hunk 3 above)
Comment (ediblecode) [reply packages/gatsby/index.d.ts]: Good spot. I just capitalised the name and didn't follow it through but you're right!

Comment (pieh) [reply packages/gatsby/index.d.ts]: Oh, yup! You are not introducing wrong args here - it just clicked when reviewing PR that what we currently have is wrong

File: packages/gatsby/index.d.ts (hunk 4 above)
Comment (muescha) [reply packages/gatsby/index.d.ts]: @pieh 
there is also an:

```javascript

/**
 * If you want to strip the initial indentation from the beginning of each line in a multiline string.
 * Important note: this tag will not indent multiline strings coming from the substitutions.
 * If you want that behavior, use the `html` tag (aliases: `source`, `codeBlock`).
 */
 export const stripIndent: TemplateTag;


```

**Comment by LekoArts** [comment]
Hi @ediblecode 👋 Do you have time to address the requested changes?

**Comment by ediblecode** [comment]
@pieh no worries, I enjoyed it. Thanks for the review. Glad to help out.

Also, worth saying the reporter stuff for showing progress when devving a source plugin is lovely!

**Comment by pieh** [comment]
Published `gatsby@2.18.1` / `gatsby-cli@2.8.13`

**Pull request merged**